source/common/modules/markdown-editor/util/remove-markdown-link.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        removeMarkdownLink function
+ * CVM-Role:        Utility function
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This function removes a markdown link, performing necessary
+ *                  transformations where applicable.
+ *
+ * END HEADER
+ */
+
+import { type EditorView } from '@codemirror/view'
+import { type ChangeSpec } from '@codemirror/state'
+import { type SyntaxNode } from '@lezer/common'
+
+/**
+ * Utility to remove a markdown link. It extracts the text from the markdown
+ * link and replaces the link with just the text.
+ *
+ * @param {EditorView} view - The editor view
+ * @param {SyntaxNode} node - The node containing the Link
+ */
+export function removeMarkdownLink (node: SyntaxNode, view: EditorView): void {
+  // To deal with user being able to click on either part of the link (Text or URL)
+  // And the two different types of links []() and <>
+  // Check which node they clicked on before passing it
+  var useNode: SyntaxNode = node
+  if (node.type.name === 'URL' && node.parent != null) {
+    if (node.parent.type.name === 'Link') {
+      useNode = node.parent
+    } if (node.parent.type.name === 'Paragraph') {
+      useNode = node
+    }
+  }
+
+  const from = useNode.from
+  const to = useNode.to
+
+  var linkText: string
+  if (useNode.type.name === 'URL') {
+    // This is where the URL with <> formate is parsed
+    // We add +- 1 to remove the < > brackets from the text
+    linkText = view.state.sliceDoc(node.from+1, node.to-1)
```

Comment:
Can you be certain that the first and last characters will always be `<` or `>`, respectively? There may be other types of links that the parser understands that we don't yet account for.

Reply:
I found an existing function: shortenUrlVisually (/source/common/util/shorten-url-visually.ts) which handles what I am doing here, should I utilize this function instead?

Reply:
Please don't — `shortenUrlvisually` should only be used in contexts where what it spits out doesn't have to be a functioning url. As it says in the name: It's for visual purposes. Please just copy over whatever you need from that function here to prevent this from breaking.